docs/tutorials/podman-socket-activation.md

```diff
@@ -0,0 +1,208 @@
+sockets that the container should use
+
+```
+[Unit]
+Description=echo server
+
+[Socket]
+ListenStream=127.0.0.1:3000
+ListenDatagram=127.0.0.1:3000
+ListenStream=[::1]:3000
+ListenDatagram=[::1]:3000
+ListenStream=%h/echo_stream_sock
+
+# VMADDR_CID_ANY (-1U) = 2^32 -1 = 4294967295
+# See "man vsock"
+ListenStream=vsock:4294967295:3000
+
+[Install]
+WantedBy=default.target
+```
+
+`%h` is a systemd specifier that expands to the user's home directory.
+
+After editing the unit files, systemd needs to reload it's configuration
+
+```
+$ systemctl --user daemon-reload
+```
+
+Start the socket unit
+
+```
+$ systemctl --user start echo.socket
+```
+
+Test the echo server with the program __socat__
+
+```
+$ echo hello | socat - tcp4:127.0.0.1:3000
+hello
+$ echo hello | socat - tcp6:[::1]:3000
+hello
+$ echo hello | socat - udp4:127.0.0.1:3000
+hello
+$ echo hello | socat - udp6:[::1]:3000
+hello
+$ echo hello | socat - unix:$HOME/echo_stream_sock
+hello
+$ echo hello | socat - VSOCK-CONNECT:1:3000
+hello
+```
+
+The echo server works as expected. It replies _"hello"_ after receiving the text _"hello"_.
+
+### Socket activate an Apache HTTP server with systemd-socket-activate
+
+Instead of setting up a systemd service to test out socket activation, an alternative is to use the command-line
+tool [__systemd-socket-activate__](https://www.freedesktop.org/software/systemd/man/systemd-socket-activate.html#).
+
+As an example let us use the container image
+[ghcr.io/eriksjolund/socket-activate-httpd](https://github.com/eriksjolund/socket-activate-httpd/pkgs/container/socket-activate-httpd)
+that contains an Apache HTTP server.
+
+In one shell, start __systemd-socket-activate__.
+
+```
+$ systemd-socket-activate -l 8080 podman run --rm --network=none ghcr.io/eriksjolund/socket-activate-httpd
+```
+
+The TCP port number 8080 is given as an option to __systemd-socket-activate__. The  __--publish__ (__-p__)
+option for `podman run` is not used.
+
+In another shell, fetch a web page from _localhost:8080_
+
+```
+$ curl -s localhost:8080 | head -6
+<!doctype html>
+<html>
+  <head>
+<meta charset='utf-8'>
+<meta name='viewport' content='width=device-width, initial-scale=1'>
+<title>Test Page for the HTTP Server on Fedora</title>
+$
+```
+
+### Network throughput and latency
+
+Using _socket activation_ comes with another advantage. The communication in the socket-activated
+socket has native network throughput. When running rootless Podman, other network traffic needs to pass
+through slirp4netns and gets the performance penalty that comes with it.
+
+Unfortunately, using _socket activation_ also comes with a disadvantage. The very first connection
+to a socket-activated container will have more latency due to container startup.
+To minimize this latency, consider adding the __podman run__ option __--pull=never__ and
+instead pull the container image beforehand.
+
+### Note about SELinux
+
+> **Note**
```

Comment:
I'm not sure I like how the "Note" resolves.  I think I'd drop it.

Reply:
I removed both notes. The importance of the first note will eventually fade away as people are using newer versions of container-selinux.

The second note is probably more of interest to the end users that are super interested and not to the general end user. 

```
### Note about SELinux

> **Note**
> If your computer is running SELinux, you need to have [__container-selinux 2.186.0__](https://github.com/containers/container-selinux)
> or newer installed. If container socket activation via Podman does not work and you are using an older version of
> container-selinux, add `--security-opt label=disable` to `podman run` as a work around.

### Note about running systemd in container

> **Note** There is currently no way to pass a socket-activated socket into a container when the container is running
> systemd. In other words, the executable __/usr/lib/systemd/systemd__ does not support socket activation.
> (There is a feature request related to it https://github.com/systemd/systemd/issues/17764)
```